packages/java/src/browser/java-decorator.ts

```diff
@@ -0,0 +1,82 @@
+/********************************************************************************
+ * Copyright (C) 2018 RedHat and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable, postConstruct } from 'inversify';
+import { Event, Emitter } from '@theia/core/lib/common/event';
+import { Tree, TreeNode } from '@theia/core/lib/browser/tree/tree';
+import { DepthFirstTreeIterator } from '@theia/core/lib/browser/tree/tree-iterator';
+import { TreeDecorator, TreeDecoration } from '@theia/core/lib/browser/tree/tree-decorator';
+import { EditorManager } from '@theia/editor/lib/browser';
+import { JAVA_LANGUAGE_ID } from '../common';
+
+@injectable()
+export class JavaDecorator implements TreeDecorator {
```

Comment:
I agree with @kittaakos that use of "details" should not be Java-specific. See my comment in the PR main page for more.